### ISSUE TITLE
Get-MgUser does not return extension attributes unless you specifically name them

### ISSUE THREAD

**Issue by cyr0nk0r** [microsoftgraph/msgraph-sdk-powershell#1929]
`Get-AzureADUser -ObjectId user@example.com | select -ExpandProperty ExtensionProperty`
Returns
![1](https://user-images.githubusercontent.com/77940332/230758707-83710d9b-0f6e-4c17-b085-d2e5756d9c1f.png)
In this example, we only have 1 AAD extension attribute (the info field), but other environments might have many more. If we had more than 1, the above command would list all the extension attributes for a user that aren't null.

`(get-mguser -UserId user@example.com).AdditionalProperties`
Returns
![2](https://user-images.githubusercontent.com/77940332/230758733-892ecb02-b31e-47ad-ae99-904c0a91dc2c.png)
This should return the same information, but it doesn't.

`(get-mguser -UserId user@example.com -Property extension_<tenant>_info).AdditionalProperties`
Returns
![3](https://user-images.githubusercontent.com/77940332/230758764-5df4018a-3218-4b05-ba75-028f196d944f.png)
As you can see, when querying using Get-MgUser it will not return AAD extension attributes unless you specifically query the EXACT property you want to include. This is not ideal, because maybe you don't know the tenant ID, maybe you don't know which attribute you want to see, etc.

@BlckRo's workaround works (https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/236#issuecomment-824823836), but you shouldn't have to specify the exact attribute name to have it show up. You should be able to expand 'AdditionalProperties' or some other method to get a list of all extension attributes that a user has without having to specifically specify each and every one.

_Originally posted by @cyr0nk0r in https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/236#issuecomment-1501056915_

**Comment by SeniorConsulting** [comment]
Gidday cyr0nk0r, if you try to use the Graph Explorer, you'll notice that the same behaviour is experienced. This is the way that the Graph API was designed, so the PowerShell modules can only replicate the same behaviour.

I'd recommend using the beta endpoint (if what you're working on is non-production), which generally retreives all properties on the first hit. This will allow you to see the names of all of the properties returned, upon which point you can switch back to the v1.0 version of the API and run those commands including the property.

If you're not using the pre-release version of the Microsoft.Graph PowerShell modules, then you can complete something similar to the following:
```
Connect-MgGraph
Select-MgProfile beta
$TestUser = Get-MgUser -UserId Test.User@contoso.com
$TestUser.AdditionalProperties

**Comment by cyr0nk0r** [comment]
You are correct that selecting the beta profile returns expect additional extension attributes.
If that is the case, is it just a matter of time for the beta functionality to merge into the production functionality of get-mguser?
Will this issue sort itself out? If it works in beta, do I just need to wait?

**Comment by SeniorConsulting** [comment]
I'm not a representative from Microsoft, so I can't really speak from that perspective. However, I doubt that this functionality will be merged into v1.0 of the Graph API - mainly because always getting all properties will result in a lot of extra overhead (sending data back to anyone who requests it, especially if you're requesting all users from your tenant).

I suppose you could check the feedback portal and see if anyone else has requested this feature from the feedback portal, and if not, raise your own feature request:
https://feedbackportal.microsoft.com/feedback/forum/ebe2edae-97d1-ec11-a7b5-0022481f3c80

It looks like you can view a Graph API changelog here:
https://developer.microsoft.com/en-us/graph/changelog/

**Comment by peombwa** [comment]
As stated above, this is by design when targeting v1.0. The v1.0 endpoint will only return directory extensions on `$select` (`-Property`). See the API docs on directory extensions at https://learn.microsoft.com/en-us/graph/extensibility-overview?tabs=http#retrieve-a-directory-extension for more details:
> The extension property will be returned by default through the beta endpoint, but only on $select through the v1.0 endpoint.

Please use the forum link shared above to upvote/create a service feature request for your scenario.
> https://feedbackportal.microsoft.com/feedback/forum/ebe2edae-97d1-ec11-a7b5-0022481f3c80
